rewrite-maven/src/main/java/org/openrewrite/maven/tree/ResolvedPom.java

```diff
@@ -470,6 +487,18 @@ private void mergeRequestedDependencies(List<Dependency> incomingRequestedDepend
             }
         }
 
+        private void mergePlugins(List<Plugin> incomingPlugins) {
+            if (!incomingPlugins.isEmpty()) {
+                if (plugins == null || plugins.isEmpty()) {
+                    //It is possible for the plugins to be an empty, immutable list.
+                    //If it's empty, we ensure to create a mutable list.
+                    plugins = new ArrayList<>(incomingPlugins);
+                } else {
+                    plugins.addAll(incomingPlugins);
+                }
+            }
+        }
```

Comment:
What would happen if the current project and a parent both have a `maven-compiler-plugin`? Are these then both added to this list?

What about a plugin that occurs both in `<build><pluginManagement>` and `<build><plugins>` with configuration in both?

Reply:
oh, it seems that there is an interesting fine-tunning in inheritance of plugins from parent: https://www.baeldung.com/maven-plugin-override-parent.

Reply:
Implemented some merging logic. Missing merging of configurations and executions.
For configurations I think we won't be able to do the fine-grained merging, since we are missing the xml attributes in the JsonNode object.
For executions still not clear to me how to merge them exactly.

Reply:
@timtebeek is there any reason not to merge this as-is and then work on the better merging?

Reply:
Ok to merge; I'd held off as every change to tree classes requires a coordinated deploy of the platform, right?
Figured best to limit the amount of coordinated releases and deployments, but haven't found time to dive deeper here.